Update Visual Studio build with CMake

VOM7HC opened the pull request:
- Update README setup Windows build
- Add CMakeSettings.json for Visual Studio CMake. Now project can open, build, debug, profiling directly in Visual Studio 2019 and no need CMake generated solution and vcxproj anymore.
- Remove obsolete Visual Studio solution and vcxproj, which can not build.
- Switch to CMakeSettings.json mean support cross compile Linux in Visual Studio too. [From Visual Studio 2019, MS support debug and release Linux build same CMakeList.txt](https://docs.microsoft.com/en-us/cpp/linux/cmake-linux-project?view=msvc-160). 
NOTE: I just checked my commit on Windows, not test Linux.

VOM7HC committed:
Update for Visual Studio project:
+ Add CMakeSettings.json for Visual Studio CMake
- Remove obsolete Visual Studio solution and project
! Update README.md

VOM7HC committed:
Update README doc

VOM7HC commented:
Hi Victor,
For Linux support in Visual Studio, I need you help merge Windows branch to master branch. I willing help *NIX support in VS when you complete merge.

WesleyKhoiVo commented:
Please wait, I need commit another code to merge

VOM7HC committed:
* Update for Visual Studio project:
+ Add Visual Studio Code support for Windows
! Update README.md

VOM7HC commented:
Hi @vi3itor, I added support Visual Studio 2019 IDE and Visual Studio Code editor,
- When you have a free time, can you build it and recheck again?
- With VS code support. In theory, it will help build Mac and Linux too. But I just added Windows - MSVC and I can't check when it still be 2 separate branch.

VOM7HC commented:
And Vcpkg is cross platform, may be switch to one Vcpkg is better than use Vcpkg - brew - Linux repository

vi3itor commented:
Hi Khoi,

Thanks again for the contributions! I was busy with work-related projects last week but I've seen your commits and discussion  in other issues. I will check the PR and think how to integrate it into existing codebase, considering that I'll soon merge `windows` into `main` branch and use vcpkg manifests at least on Windows. I tried vcpkg on Mac two weeks ago, but it failed to install some SDL libraries.
I'll get back to you in a couple of days!

vi3itor commented on .vscode/c_cpp_properties.json:
```diff
@@ -0,0 +1,18 @@
+                "${vcpkgRoot}/x86-windows/include"
+            ],
+            "defines": [],
+            "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x86/cl.exe",
```
We should probably make it an empty string not to rely on a particular version of the compiler:
https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference

VOM7HC replied:
Thanks, I will make another commit for it

vi3itor commented on README.md:
```diff
@@ -103,29 +102,76 @@ cd GameplayFootball
+![CMake tool](https://docs.microsoft.com/en-us/cpp/build/media/cmake-install-2019.png?view=msvc-160)
+
+Build project
+- Start `Visual Studio 2019`, then click `File > Open > Folder...`
+- Navigate to the `GameplayFootball` project folder, and click `Select Folder`
+- Select `x86-Debug` or `x86-ReleaseDebugSymbol` and `x86-Release` mode in drop down menu `Manage Configurations...`
+
+![Manage Configurations...](https://docs.microsoft.com/en-us/visualstudio/ide/media/understanding-build-configurations/active-config.png?view=vs-2019)
+- Click `gameplayfootball.exe` in menu `Select startup item...` to start build game
+
+![Select startup item...](https://docs.microsoft.com/en-us/cpp/build/media/cmake-run-button.png?view=msvc-160)
+- After build success, game will start but it close immediately, because it missing data. Now go to step copy data to game
+
+To copy data to game, switch back to Command Prompt window
```
I like the way you extensively comment on setting up the project in Visual Studio and VS Code. I'm even thinking of moving it to a separate file (like `development.md`) and adding similar instructions for other platforms/tools once I merge branches. But let's leave it as it is for now.

VOM7HC replied:
Hi @vi3itor, some "new" VS 2019 features will make old user or newbie (who never use VS and use old document for reference) is hard to use, so I add image for easier to use it

vi3itor commented on .vscode/settings.json:
```diff
@@ -0,0 +1,6 @@
+{
+    "C_Cpp.errorSquiggles": "Enabled",
+    "cmake.configureSettings": {
+        "CMAKE_TOOLCHAIN_FILE": "C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake"
```
We should use `${vcpkgRoot}` in case the installation path is different. (Same as in `c_cpp_properties.json`)

VOM7HC replied:
Hi @vi3itor,  `${vcpkgRoot}` is return `C:/dev/vcpkg/installed` instead of `C:/dev/vcpkg`, do you have any idea?
![image](https://user-images.githubusercontent.com/20337876/129718106-66bbaf6e-9ffa-48e3-bd83-ae66d1ec25f9.png)

vi3itor replied:
Hi @WesleyKhoiVo, it looks like the variable is [only works with the C++ extension](https://github.com/microsoft/vscode-cpptools/issues/4730). We probably should ask the user to set `VCPKG_ROOT` variable once and use it everywhere (similar to https://github.com/vi3itor/football/blame/windows/gfootball/doc/compile_engine.md#L46)

vi3itor reviewed (changes requested):
Everything looks good to me. Could you please make minor adjustments to VS Code settings and properties files and check if it works correctly in the IDE? Once it's ready, I'll merge the changes.

VOM7HC committed:
* Update for Visual Studio project:
! Use system variable VCPKG_ROOT for Vcpkg path
! Update README.md for VS code instruction compiler
+ Add Clang-tidy for VS 2019

VOM7HC committed:
* Update for Visual Studio project:
! Update readme.md for instruction VCPKG_ROOT

VOM7HC commented:
Hi Victor, please review again my PR. Thanks

vi3itor commented:
I'll check in tomorrow.
